packages/appframe/src/react/index.js

```diff
@@ -0,0 +1,303 @@
+
+function SkipBanner(props) {
+  return (
+    <Theme name={Theme.names.dark}>
+      <div {...styles.skipBanner()} {...props}>
+        <Button
+          appearance={Button.appearances.secondary}
+          size={Button.sizes.small}
+          tabIndex={0}
+        >
+          skip to main content
+        </Button>
+      </div>
+    </Theme>
+  )
+}
+
+SkipBanner.propTypes = {
+  href: PropTypes.string.isRequired
+}
+
+const SkipTarget = React.forwardRef((props, ref) => {
+  return <a ref={ref} tabIndex={-1} {...props} />
+})
+
+SkipTarget.displayName = 'SkipTarget'
+SkipTarget.propTypes = {
+  id: PropTypes.string.isRequired
+}
+
+function Container(props) {
+  const context = useContext(AppFrameContext)
+  return <div {...styles.container(context.sidenavVariant)} {...props} />
+}
+
+function SideNav(props) {
+  const { sidenavVariants: variants } = vars
+
+  const { children, ...rest } = props
+  const { closeSidenav, openSidenav, sidenavVariant } = useContext(
+    AppFrameContext
+  )
+
+  const hoverable = sidenavVariant === variants.minimized
+  const [hovered, setHovered] = useState(false)
+
+  const variant = hoverable && hovered ? variants.overlay : sidenavVariant
+  const visible = variant === variants.overlay || variant === variants.open
+
+  const ref = React.useRef()
+
+  const overlayed = variant === variants.overlay
+  useBodyScrollLock(ref, overlayed)
+
+  const closeIfOverlayed = useCallback(() => {
+    if (overlayed) closeSidenav()
+  }, [overlayed, closeSidenav])
+
+  useOnClickOutside(ref, evt => {
+    const inTopNav = evt.target && !!evt.target.closest(`#${TOP_NAV_ID}`)
+    if (!inTopNav) closeIfOverlayed()
+  })
+  useOnEscape(closeIfOverlayed)
+  useOnInnerFocus(ref, { onEnter: openSidenav, onLeave: closeIfOverlayed })
+
+  return (
+    <Theme name={Theme.names.dark}>
+      <div
+        ref={ref}
+        {...styles.sidenav(variant)}
+        {...rest}
+        {...(hoverable && {
+          onMouseEnter: () => setHovered(true),
+          onMouseLeave: () => setHovered(false)
+        })}
+      >
+        <div {...styles.sidenavOverflowMask()}>
+          <Scrollable {...styles.sidenavInner()}>
+            {isFunction(children) ? children({ visible }) : children}
+          </Scrollable>
+        </div>
+      </div>
+    </Theme>
+  )
+}
+
+SideNav.propTypes = {
+  children: PropTypes.oneOfType([PropTypes.func, PropTypes.node])
+}
+SideNav.variants = vars.sidenavVariants
+
+function TopNav(props) {
+  const { children, ...rest } = props
+  const { closeSidenav, openSidenav, sidenavOpen } = useContext(AppFrameContext)
+
+  const meta = { closeSidenav, openSidenav, sidenavOpen }
+
+  return (
+    <Theme name={Theme.names.dark}>
+      <div data-ps-appframe-topnav {...styles.topnav()} {...rest}>
+        {isFunction(children) ? children(meta) : children}
+      </div>
+    </Theme>
+  )
+}
+TopNav.propTypes = {
+  children: PropTypes.oneOfType([PropTypes.func, PropTypes.node])
+}
+
+AppFrame.SideNav = SideNav
```

Comment:
There might be some organizational potential here by extracting some of the smaller components.